src/ja/server/dispatcher/Dispatcher.py

```diff
@@ -0,0 +1,43 @@
+from ja.common.job import Job
+from ja.server.proxy.proxy import WorkerProxy
+from typing import List
+from ja.server.database.database import ServerDatabase
+
+
+JobDistribution = ServerDatabase.JobDistribution
+
+
+class Dispatcher:
+    """
+    Represents the mediator between the Scheduler and the work machines.
+    """
+
+    def __init__(self, worker_proxies: List[WorkerProxy]):
```

Comment:
The list of proxies may change because new machines can be added and removed at any time.

Here's the solution that we thought of: There should be another class, perhaps with the name `WorkerProxyFactory`. It has one method which is `get_proxy` and takes a parameter the WorkMachine from the database. It returns a WorkerProxy for this WorkMachine.

This WorkerProxyFactory class also cleans up work machines which are no longer connected (this should be just mentioned in the class description, no public methods for this)

Reply:
would it be wrong if we use a setter for the list of proxies, or just give each instance of the work machine a proxy variable that we can access, or store the proxy details in the database and access it from there

Reply:
The connection details are already available in the WorkMachine (like IP address). However, we cannot store the Proxy object in the database (because it is a Python object, not some value).

We could have a setter for proxies, but I think it would be better if we can have an AbstractFactory which simply gives back the appropriate proxy.

If you are unsure about what I mean, I could send it in another pull request, it is a simple addition.

Reply:
still not sure why we need to make another class that returns only 1 variable that we can just store in the work machine class

Reply:
The other class will also manage the internal cache of proxies, and also handle work machine disconnecting (it needs to update database in that case)